pkg/devfile/parser/parse.go

```diff
@@ -392,10 +397,38 @@ func parseFromURI(importReference v1.ImportReference, curDevfileCtx devfileCtx.D
 	if !absoluteURL && curDevfileCtx.GetAbsPath() != "" {
 		newUri = path.Join(path.Dir(curDevfileCtx.GetAbsPath()), uri)
 		d.Ctx = devfileCtx.NewDevfileCtx(newUri)
+		if !strings.HasSuffix(newUri, ".yaml") {
+			return DevfileObj{}, fmt.Errorf("the provided path is not a valid yaml filepath: %s", newUri)
+		}
+		if util.ValidateFile(newUri) != nil {
+			return DevfileObj{}, fmt.Errorf("failed to read devfile from path %s", newUri)
+		}
+		srcDir := path.Dir(newUri)
+		destDir := path.Dir(curDevfileCtx.GetAbsPath())
+		if srcDir != destDir {
+			err := util.CopyAllDirFiles(srcDir, destDir)
+			if err != nil {
+				return DevfileObj{}, err
+			}
+		}
 	} else if absoluteURL {
 		// absolute URL address
 		newUri = uri
 		d.Ctx = devfileCtx.NewURLDevfileCtx(newUri)
+		if strings.Contains(newUri, "raw.githubusercontent.com") {
+			urlComponents, err := util.GetGitUrlComponentsFromRaw(newUri)
+			if err != nil {
+				return DevfileObj{}, err
+			}
+			if !strings.Contains(urlComponents["file"], "devfile.yaml") {
+				return DevfileObj{}, fmt.Errorf("%s is not a devfile", newUri)
+			}
+			destDir := path.Dir(curDevfileCtx.GetAbsPath())
+			err = getStackFromGit(urlComponents, destDir)
+			if err != nil {
+				return DevfileObj{}, err
+			}
+		}
 	} else if curDevfileCtx.GetURL() != "" {
```

Comment:
what about this case? consider a nested parent reference, for example
main devfile has a parent reference `somegithuburl.devfile.yaml`
and the 1st levell parent devfile from `somegithuburl.devfile.yaml` also has a second level parent reference `subfolder/devfile.yaml`.

Reply:
It looks like this scenario is already covered:

```yaml
# main devfile
parent:
    uri: https://raw.githubusercontent.com/mike-hoang/registry-support/API-46/tests/registry/stacks/nodejs-parent/devfile.yaml
```

```yaml
# nodejs-parent/devfile.yaml
# contains file: parent/testing.txt
parent:
    uri: ./parent/devfile.yaml
```

dest dir contains testing.txt after parsing

Reply:
this case works is because your 2nd parent is under the same level (or cases under lower level sub-dirs also will work), and when parsing the first parent, it is already downloaded as first parent's resource. 

if the 2nd parent is from a upper level dir, the resources are not going to be downloaded. i.e.
```
first parent: somegiturl.com/nodejs-parent/devfile.yaml
second parent: somegiturl.com/nested-parent/devfile.yaml


parent:
    uri: ../nested-parent/devfile.yaml
```